internal/core/src/exec/expression/LogicalUnaryExpr.cpp

```diff
@@ -0,0 +1,39 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "LogicalUnaryExpr.h"
+
+namespace milvus {
+namespace exec {
+
+void
+PhyLogicalUnaryExpr::Eval(EvalCtx& context, VectorPtr& result) {
+    AssertInfo(inputs_.size() == 1,
+               fmt::format("logical unary expr must has one input, but now {}",
+                           inputs_.size()));
+
+    inputs_[0]->Eval(context, result);
+    if (expr_->op_type_ == milvus::expr::LogicalUnaryExpr::OpType::LogicalNot) {
+        auto flat_vec = GetFlatVector(result);
+        bool* data = static_cast<bool*>(flat_vec->GetRawData());
+        for (int i = 0; i < flat_vec->size(); ++i) {
+            data[i] = !data[i];
+        }
+    }
```

Comment:
Throw exception if op_type is not `LogicalNot`, in case we did the wrong parse, though it's not very possible.

Reply:
It's not a good ideal ,because this is computing process. and if check in here, will have bad effect in performance,  parser and precheck should prove it is LogicalNot.

Reply:
OK. But I noticed we have checked the op type in `LogicalBinary`.